x-pack/legacy/plugins/uptime/public/lib/adapters/source_settings/get_source_settings.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import axios from 'axios';
+import { getApiPath } from '../../helper';
+
+/**
+ * Fetches and returns the uptime index pattern, optionally provides it to
+ * a given setter function.
+ * @param basePath - the base path, if any
+ * @param setter - a callback for use with non-async functions like `useEffect`
+ */
+export const getSourceSettings = async (basePath?: string, setter?: (data: unknown) => void) => {
+  try {
+    const { data } = await axios.get(getApiPath('/api/uptime/source_settings', basePath));
```

Comment:
Is there a reason not to use http.fetch instead? Our Security model heavily depends on the fact that every plugin uses core provided API

Reply:
Thanks for the pointer, will make the switch, this file was copied from some much older code pre NP

Reply:
I think we need to do this across the whole app. We're tracking that here. I'll leave this PR as-is since we have to make a change everywhere ultimately.